client/app/controllers/collection/add.js

```diff
@@ -68,99 +68,99 @@ export default Ember.Controller.extend({
+                        if (typeof actionObj.then === 'string') {
+                            fireActions.call(this, actionObj.then);
                         }
-                        action_obj.output_parameter.value = result;
-                        action_obj.output_parameter.state = ['defined'];
+                        actionObj.outputParameter.value = result;
+                        actionObj.outputParameter.state = ['defined'];
                         controller.notifyPropertyChange('parameters');
                     }
                 }
             }
 
-            fire_actions.call(this, action.id);
+            fireActions.call(this, action.id);
         });
     },
 
 
     // Take the description of an action and set its properties to be the vaious literal
     // functions and parameters it depends on to operate.
-    hydrate_action: function(action) {
-        const parameters = this.get('parameters');
-        if (typeof parameters[action.output_parameter] !== 'object') {
-            parameters[action.output_parameter] = {};
+    hydrateAction: function(action) {
+        const PARAMETERS = this.get('parameters');
+        if (typeof PARAMETERS[action.outputParameter] !== 'object') {
+            PARAMETERS[action.outputParameter] = {};
         }
         if (typeof action.parameters !== 'object') {
             action.parameters = {};
         }
+        const SIGNATURE = this.get(action.type + 'Signature');
         // Create a new object as not to modify the object returned from the model
-        const hydrated_action = {
+        const HYDRATED_ACTION = {
             id: action.id,
             type: action.type,
-            signature: this.get(action.type + '_signature'),
+            signature: SIGNATURE,
             action: this.get(action.type),
             conditions: action.conditions,
             parameters: Object.keys(action.parameters).reduce((result, key) => {
-                if (typeof parameters[action.parameters[key]] !== 'object') {
-                    parameters[action.parameters[key]] = {
+                if (typeof PARAMETERS[action.parameters[key]] !== 'object') {
+                    PARAMETERS[action.parameters[key]] = {
                         state: ['undefined'],
                         value: undefined
                     };
                 }
-                result[key] = parameters[action.parameters[key]]
+                result[key] = PARAMETERS[action.parameters[key]]
                 return result;
             }, {}),
             args: action.args,
-            output_parameter: parameters[action.output_parameter],
+            outputParameter: PARAMETERS[action.outputParameter],
             then: action.then
         };
-        hydrated_action['arg_arr'] = constructArgArr.call(this, hydrated_action);
-        return hydrated_action;
+        HYDRATED_ACTION['argArr'] = constructArgArr.call(this, HYDRATED_ACTION);
```

Comment:
'argArr' sounds like not the most descriptive name, can you think of alternatives?